### ISSUE TITLE
Modularise tfjs core ops (tracking issue)

### ISSUE THREAD

**Issue by tafsiri** [tensorflow/tfjs#2822]
## Op Files

The files below are all in `tfjs-core/src/ops` and contain ops to be modularized. They are in alphabetical order.

  - [x] array_ops.ts (ann, yannick)
  - [x] batchnorm.ts (na)
  - [x] binary_ops.ts (yannick)
  - [x] boolean_mask.ts
  - [x] browser.ts
  - [x] compare_ops.ts (na)
  - [x] complex_ops.ts (yannick)
  - [x] concat_split.ts (na)
  - [x] confusion_matrix.ts
  - [x] conv.ts (na)
  - [x] diag.ts (na)
  - [x] dropout.ts
  - [x] fused_ops.ts (yannick)
  - [x] gather_nd.ts (yannick)
  - [x] image_ops.ts (yannick)
  - [x] in_top_k.ts
  - [x] linalg_ops.ts (yannick)
  - [x] logical_ops.ts (yannick)
  - [x] loss_ops.ts (yannick)
  - [x] lrn.ts (na)
  - [x] lstm.ts (yannick)
  - [x] matmul.ts (na)
  - [x] moving_average.ts (yannick)
  - [x] norm.ts (yannick)
  - [x] pool.ts (na)
  - [x] reduction_ops.ts (yannick)
  - [x] relu_ops.ts (na)
  - [x] reverse.ts (yannick)
  - [x] scatter_nd.ts (yannick)
  - [x] segment_ops.ts (yannick)
  - [x] signal_ops.ts (yannick)
  - [x] slice.ts (yannick)
  - [x] softmax.ts (yannick)
  - [x] sparse_to_dense.ts (yannick)
  - [x] spectral_ops.ts (ann)
  - [x] strided_slice.ts (yannick)
  - [x] tensor_ops.ts (yannick)
  - [x] topk.ts (yannick)
  - [x] transpose.ts (yannick)
  - [x] unary_ops.ts (yannick)

## Kernel Names
(List generated from backend.ts)

- batchMatMul
- fusedBatchMatMul
- slice
- stridedSlice
- unstack
- reverse
- concat
- neg
- add
- addN
- subtract
- multiply
- realDivide
- divNoNan
- floorDiv
- sum
- prod
- unsortedSegmentSum
- argMin
- argMax
- equal
- notEqual
- less
- lessEqual
- greater
- greaterEqual
- logicalNot
- logicalAnd
- logicalOr
- where
- select
- topk
- min
- minimum
- mod
- max
- maximum
- all
- any
- squaredDifference
- ceil
- floor
- round
- sign
- isNaN
- isInf
- isFinite
- pow
- exp
- expm1
- softmax
- log
- log1p
- sqrt
- rsqrt
- square
- reciprocal
- relu
- relu6
- prelu
- elu
- eluDer
- selu
- int
- clip
- abs
- complexAbs
- sigmoid
- softplus
- sin
- cos
- tan
- asin
- acos
- atan
- atan2
- sinh
- cosh
- tanh
- asinh
- acosh
- atanh
- erf
- step
- fusedConv2d
- conv2d
- conv2dDerInput
- conv2dDerFilter
- fusedDepthwiseConv2D
- depthwiseConv2D
- depthwiseConv2DDerInput
- depthwiseConv2DDerFilter
- conv3d
- conv3dDerInput
- conv3dDerFilter
- maxPool
- maxPoolBackprop
- avgPool
- avgPoolBackprop
- avgPool3d
- avgPool3dBackprop
- maxPool3d
- maxPool3dBackprop
- reshape
- cast
- tile
- pad
- transpose
- gather
- gatherND
- scatterND
- batchToSpaceND
- spaceToBatchND
- resizeBilinear
- resizeBilinearBackprop
- resizeNearestNeighbor
- resizeNearestNeighborBackprop
- batchNormalization
- localResponseNormalization4D
- LRNGrad
- multinomial
- oneHot
- cumsum
- nonMaxSuppression
- fft
- ifft
- complex
- real
- imag
- cropAndResize
- depthToSpace
- split
- sparseToDense
- diag
- fill
- onesLike
- zerosLike
- linspace
- dispose

## New Kernels
Kernels that did not exist previously on the backend interface that need to be added as kernels in the modularized backends.

- BroadcastTo 
- Identity

**Comment by tafsiri** [comment]
@annxingyuan could you put your name next to the kernels you are already working on for softmax.

**Comment by sacredvoid** [comment]
Hey, can we expect support for 'depthToSpace' kernel on the WASM backend anytime soon?

**Comment by tafsiri** [comment]
@sacredvoid could you file a new issue for that as a feature request. It will be easier to track as its own issue.
